packages/lesswrong/server/autocompleteEndpoint.ts

```diff
@@ -73,21 +74,25 @@ async function constructMessageHistory(
 ): Promise<PromptCachingBetaMessageParam[]> {
   const messages: PromptCachingBetaMessageParam[] = [];
 
-  // Make the fetches parallel to save time
-  const [posts, comments] = await Promise.all([
-    runFragmentQuery({
+  const posts = await getWithCustomLoader(context, "postsHistoryForLLM", postIds, async (postIdss: string[][]) => {
```

Comment:
using an array (`postIds`) for a key is pretty sus here. prolly I should just write a `getManyWithCustomLoader` helper

Reply:
Is it sus?  `getWithCustomLoader` explicitly accepts arbitrary ID types (and a corresponding `idsToResults`)

Reply:
I mean, it's a bit of a mess because of the fragility of reference equality